src/cascade/executor/create_settings.py

```diff
@@ -211,45 +258,100 @@ def create_settings(rng, locations):
+    # Only add to dict if there are some?
+    if random_effects:
+        case["random_effect"] = random_effects
 
 
-def add_random_effects(children, has, rate, rng):
-    random_effects = list()
-    if children:
-        for random_effect, exists in [(x, y) for (x, y) in has.items() if y]:
-            single_age = rate == "pini"
-            choice = rng.choice(["none", "all", "every"])
-            if choice == "all":
-                grid_case = grid(False, rng, single_age)
-                grid_case["rate"] = random_effect
-                grid_case["location"] = 1
-                random_effects.append(grid_case)
-            elif choice == "every":
-                for child in children:
-                    grid_case = grid(False, rng, single_age)
-                    grid_case["rate"] = random_effect
-                    grid_case["location"] = child
-                    random_effects.append(grid_case)
-    return random_effects
+def which_random_effects(nonzero_rates, rate, rng):
+    # We specify priors of random effects either
+    # a) Once for the whole hierarchy, or
+    # b) For every single location in the hierarchy.
+    # Choose this by rate
+    rate_specifies_re_by_location = dict()
+    for rate_name in nonzero_rates:
+        choice = rng.choice(["none", "all", "every"], name=f"re.{rate_name}")
+        # none means no random effects.
+        # all means every single location.
+        # every means each location gets a new one.
+        if rate in {"all", "every"}:
+            rate_specifies_re_by_location[rate_name] = choice == "every"
+        else:
+            pass  # Don't record this rate as needing a random effect.
+    return rate_specifies_re_by_location
+
+
+class SettingsChoices:
+    """
+    This class will fix certain choices to given values. Its input is
+    a list of key-value pairs, set up as for a section of a ConfigParser,
+    although this adds the section title for you.
+    """
+    def __init__(self, rng=None, settings=None):
+        if isinstance(settings, str):
+            parser = ConfigParser()
+            parser.read_string("[settings]\n" + dedent(settings))
+            self.answers = parser["settings"]
+        else:
+            self.answers = dict()
+        self.rng = rng
+        self.random = list()
 
+    def choice(self, choices, name=None, p=None):
+        if name in self.answers:
+            value = self.answers[name]
+            # ConfigParser returns strings.
+            try:
+                value = int(value)
+            except ValueError:
+                try:
+                    value = float(value)
+                except ValueError:
+                    if value.lower() == "true":
+                        return True
+                    elif value.lower() == "false":
+                        return False
+            return value
+        else:
+            assert self.rng, f"Unexpected setting chosen {name} {choices}"
```

Comment:
This assert and message combination doesn't make sense to me. This is checking if the rng is None but the message has nothing to do with that. Oh, wait. I see. This is saying that if someone doesn't supply an RNG then they can only select from the preset values. Why support that use case?

Reply:
I want to be able to make a specific JSON settings with this tool and know that I'm getting the same one every time. If I failed to specify part of it, then it will throw this error, and I'll see what I failed to specify. The use case is using this system to make one set of settings.